File: python/2533.py
while True:
    try:
        x = int(input())
        
        sumNxC = 0
        sumC = 0

        for i in range(x):
            n, c = input().split()
            sumNxC += int(n) * int(c)
            sumC += int(c)
            
        print(f'{((sumNxC)/(sumC * 100)):.4f}')
    except:
        break

File: lua/2533.lua
local function main()
    local n = tonumber(io.read())
    local sumNxC = 0
    local sumC = 0

    for i=1,n do
        local entry = io.read()
        local temp = {}
        for v in string.gmatch(entry, '[^%s]+') do
            table.insert(temp, tonumber(v))
        end
        local grade = temp[1]
        local workload = temp[2]
        sumNxC = sumNxC + (grade*workload)
        sumC = sumC + workload
    end

    local ira = sumNxC / (sumC * 100)
    print(string.format("%.4f", ira))
end

while true do
    if not pcall(main) then
        break
    end
end